astropy/units/core.py

```diff
@@ -2325,8 +2325,7 @@ def __init__(
         # kwarg `_error_check` is False, the error checking is turned
         # off.
         if _error_check:
-            if scale == 0:
-                raise UnitScaleError("cannot create a unit with a scale of 0.")
+            scale = sanitize_scale(scale)
```

Comment:
I wonder if it is better to make a new routine for this case, say `verify_scale`, and simplify `sanitize_scale` so that it only does the float and complex checks for effectively unity, real, imaginary.